sql/core/src/main/scala/org/apache/spark/sql/streaming/DataStreamWriter.scala

```diff
@@ -534,6 +554,8 @@ final class DataStreamWriter[T] private[sql](ds: Dataset[T]) {
 
   private var foreachWriter: ForeachWriter[T] = null
 
+  private var connectForeachWriter: PythonForeachWriter = null
```

Comment:
what if we declare `foreachWriter` as ForeachWriter[_]? 
I think we can remove `connectForeachWriter`. 
At line 360, we could have `val sink = ForewachWriterTable[_](...)`

Reply:
Yes that would work I think. I'll change it!